test/framework/src/main/java/org/elasticsearch/common/util/concurrent/DeterministicTaskQueue.java

```diff
@@ -200,6 +200,35 @@ public void advanceTime() {
         assert deferredTasks.isEmpty() == (nextDeferredTaskExecutionTimeMillis == Long.MAX_VALUE);
     }
 
+    public PrioritizedEsThreadPoolExecutor getPrioritizedEsThreadPoolExecutor() {
+        return getPrioritizedEsThreadPoolExecutor(Function.identity());
+    }
+
+    public PrioritizedEsThreadPoolExecutor getPrioritizedEsThreadPoolExecutor(Function<Runnable, Runnable> runnableWrapper) {
+        return new PrioritizedEsThreadPoolExecutor(
+            "DeterministicTaskQueue",
+            1,
+            1,
+            1,
+            TimeUnit.SECONDS,
+            r -> { throw new AssertionError("should not create new threads"); },
+            null,
+            null,
+            PrioritizedEsThreadPoolExecutor.StarvationWatcher.NOOP_STARVATION_WATCHER
+        ) {
+            @Override
+            public void execute(Runnable command, final TimeValue timeout, final Runnable timeoutCallback) {
+                throw new AssertionError("not implemented");
+            }
+
+            @Override
+            public void execute(Runnable command) {
+                final var wrappedCommand = runnableWrapper.apply(command);
+                runnableWrapper.apply(() -> scheduleNow(wrappedCommand)).run();
```

Comment:
I would prefer to just have an explicit log context string passed in or an explicit wrapper, seems that would be less surprising if we ever use the wrapper for anything but adding the log context.

Reply:
I know what you mean but we use a non-log-context-related wrapper in the tests - it would be a bit of a pain to assert that we wrap things correctly otherwise especially given the ongoing move away from Log4J APIs.